src/main/java/com/google/changestreams/sample/bigquery/changelog/fullschema/SchemaUtils.java

```diff
@@ -0,0 +1,539 @@
+    "_metadata_big_query_commit_timestamp";
+
+  private static final String INFORMATION_SCHEMA_TABLE_NAME = "TABLE_NAME";
+  private static final String INFORMATION_SCHEMA_COLUMN_NAME = "COLUMN_NAME";
+  private static final String INFORMATION_SCHEMA_SPANNER_TYPE = "SPANNER_TYPE";
+  private static final String INFORMATION_SCHEMA_ORDINAL_POSITION = "ORDINAL_POSITION";
+  private static final String INFORMATION_SCHEMA_CONSTRAINT_NAME = "CONSTRAINT_NAME";
+  private static final String INFORMATION_SCHEMA_ALL = "ALL";
+
+  /**
+   * @return the fields that are only used intermediately in the pipeline and are not corresponding
+   * to Spanner columns.
+   */
+  public static Set<String> getBigQueryIntermediateMetadataFieldNames() {
+    Set<String> fieldNames = new HashSet<>();
+    fieldNames.add(BQ_CHANGELOG_FIELD_NAME_ERROR);
+    fieldNames.add(BQ_CHANGELOG_FIELD_NAME_RETRY_COUNT);
+    fieldNames.add(BQ_CHANGELOG_FIELD_NAME_ORIGINAL_PAYLOAD_JSON);
+    return fieldNames;
+  }
+
+  /**
+   * @return a map where the key is the table name and the value is the SpannerTable object of the
+   * table name.
+   */
+  public static Map<String, SpannerTable> getSpannerTableByName(
+    DatabaseClient databaseClient, String changeStreamName) {
+    final Set<String> spannerTableNames = getSpannerTableNamesTrackedByChangeStreams(
+      databaseClient, changeStreamName);
+
+    final Map<String, Set<String>> spannerColumnNamesExplicitlyTrackedByChangeStreamByTableName =
+      getSpannerColumnNamesExplicitlyTrackedByChangeStreamsByTableName(
+        databaseClient, changeStreamName);
+
+    return getSpannerTableByName(databaseClient, spannerTableNames,
+      spannerColumnNamesExplicitlyTrackedByChangeStreamByTableName);
+  }
+
+  private static void appendWhereTableNamesEqualToSql(
+    StringBuilder stringBuilder, Set<String> spannerTableNames) {
+    if (!spannerTableNames.isEmpty()) {
+      stringBuilder.append(" WHERE");
+
+      for (String tableName : spannerTableNames) {
+        stringBuilder.append(" TABLE_NAME=\"");
+        stringBuilder.append(tableName);
+        stringBuilder.append("\"");
+        stringBuilder.append(" OR");
+      }
+
+      // Remove the last " OR".
+      stringBuilder.setLength(stringBuilder.length() - 3);
+    }
+  }
+
+  private static Map<String, SpannerTable> getSpannerTableByName(
+    DatabaseClient databaseClient, Set<String> spannerTableNames,
+    Map<String, Set<String>> spannerColumnNamesExplicitlyTrackedByChangeStreamByTableName) {
+
+    StringBuilder queryInformationSchemaColumnsSqlStringBuilder = new StringBuilder(
+      "SELECT TABLE_NAME, COLUMN_NAME, ORDINAL_POSITION, SPANNER_TYPE "
+        + "FROM INFORMATION_SCHEMA.COLUMNS");
+
+    // Skip the columns of the tables that are not tracked by Change Streams.
+    appendWhereTableNamesEqualToSql(queryInformationSchemaColumnsSqlStringBuilder,
+      spannerTableNames);
+
+    final ResultSet columnsResultSet =
```

Comment:
This should be in a `try with resources`

Reply:
Done.